Core/Core/Quizzes/QuizDetails/View/QuizEditorView.swift

```diff
@@ -0,0 +1,335 @@
+                accessCodeSection
+                assignmentOverridesSection
+            case .error(let errorMessage):
+                EmptyPanda(.Unsupported, title: Text("Something went wrong"), message: Text(errorMessage))
+            default:
+                Spacer()
+            }
+        }
+    }
+
+    @ViewBuilder
+    private var titleSection: some View {
+        EditorSection(label: Text("Title", bundle: .core)) {
+            CustomTextField(placeholder: Text("Add Title", bundle: .core),
+                            text: $viewModel.title,
+                            identifier: "QuizEditor.titleField",
+                            accessibilityLabel: Text("Title", bundle: .core))
+        }
+
+        EditorSection(label: Text("Description", bundle: .core)) {
+            RichContentEditor(
+                placeholder: NSLocalizedString("Add description", comment: ""),
+                a11yLabel: NSLocalizedString("Description", comment: ""),
+                html: $viewModel.description,
+                context: .course(viewModel.courseID),
+                uploadTo: .context(.course(viewModel.courseID)),
+                height: $rceHeight,
+                canSubmit: $rceCanSubmit,
+                error: Binding(get: {
+                    if case .error(let error) = alert { return error }
+                    return nil
+                }, set: {
+                    if let error = $0 { alert = .error(error) }
+                })
+            )
+                .frame(height: max(200, rceHeight))
+        }
+    }
+
+    @ViewBuilder
+    private var basicSettingsSection: some View {
+        EditorSection {
+            quizTypeRow
+            if (viewModel.shouldShowPublishedToggle) {
+                Divider()
+                ToggleRow(
+                    label: Text("Publish", bundle: .core),
+                    value: $viewModel.published)
+            }
+            if let assignmentGroup = viewModel.assignmentGroup {
+                Divider()
+                assignmentGroupRow(assignmentGroup: assignmentGroup)
+            }
+            Divider()
+            ToggleRow(
+                label: Text("Shuffle Answers", bundle: .core),
+                value: $viewModel.shuffleAnswers)
+            Divider()
+            ToggleRow(
+                label: Text("Time Limit", bundle: .core),
+                value: $viewModel.timeLimit)
+            if viewModel.timeLimit {
+                Divider()
+                DoubleFieldRow(
+                    label: Text("Length in minutes", bundle: .core),
+                    placeholder: "--",
+                    value: $viewModel.lengthInMinutes
+                )
+            }
+        }
+    }
+
+    @ViewBuilder
+    private var attemptsSection: some View {
+        EditorSection {
+            ToggleRow(
+                label: Text("Allow Multiple Attempts", bundle: .core),
+                value: $viewModel.allowMultipleAttempts)
```

Comment:
We could change the binding to `$viewModel.allowMultipleAttempts.animation()` or even change the `ToggleRow`'s initializer to `self._value = value.animation()` to make all toggles animated. I'm not sure if this latest will have any unwanted side effects on other views or if there are any other animations we want to use instead of the default animation.

Reply:
Good point! I added it to the TriggerRow. Since I introduced the whole structure here, it's not used anywhere else. If in the future we want to disable the animation, we can always add a bool parameter to the init, with default true value.